# Vector search for SOC
---

### Initialize connection to elasticsearch.

In [2]:
from dotenv import load_dotenv
from elasticsearch import Elasticsearch
import os
from pandas import json_normalize
import pandas as pd
import json
from IPython.display import display


pd.set_option('display.max_colwidth', None)

load_dotenv()
cloud_id = os.getenv('cloud_id')
api_key = os.getenv('api_key')


client = Elasticsearch(
    cloud_id=cloud_id,
    api_key=api_key
)


### query = "get object from s3 bucket named bsides zagreb"

''' Some other query examples also tried 
    query = "bucket name bsides-zagreb data exfiltration to external IP address"
    query = "bucket bsides-zagreb data leak"
    query = "get data from s3 bucket named bsides-zagreb"
''' 


response = client.search(
    index=".ds-logs-aws.cloudtrail-default-2025.03.04-000014",
    size=20,
    query={
      "knn": {
        "field": "message_secure_bert_max_inp",
        "k": 20,
        "num_candidates": 10000,
        "query_vector_builder": {
          "text_embedding": {
            "model_id": "secure-bert",
            "model_text": f"{query}"
      }
    }
  }
},
)

results = []

for hit in response["hits"]["hits"]:
    
    event_original = json.loads(hit["_source"]["event"]["original"])
    event_original['score'] = hit["_score"]
    results.append(event_original)
        
df_all = pd.DataFrame(results)
df_cleaned= df_all[['score','eventTime','userIdentity', 'eventName', 'awsRegion', "sourceIPAddress", 'userAgent', 'requestParameters', 'resources']]

display(df_cleaned)


---
    Here we can see the same query with sparse vector search using Elastics ELSER model. We get exact results ranked at top so we will continue with sparse vector search.

In [4]:

query = "get object from s3 bucket named bsides zagreb"
#query = "download object from s3 bucket from public IP" results in top 5 and IP part doesn't make the difference

response = client.search(
    index=".ds-logs-aws.cloudtrail-default-2025.03.04-000016",
    size=5,
    query={
      "sparse_vector": {
        "field": "message_elser_v2",
         "inference_id": ".elser_model_2_linux-x86_64",
          "query": query

  }
},
)

results = []

for hit in response["hits"]["hits"]:
    
    event_original = json.loads(hit["_source"]["event"]["original"])
    event_original['score'] = hit["_score"]
    results.append(event_original)

df_all = pd.DataFrame(results)
df_all['user_arn'] = df_all['userIdentity'].apply(lambda x: x['arn'])
df_cleaned= df_all[['score','eventTime','user_arn', 'eventName', 'awsRegion', "sourceIPAddress", 'userAgent', 'requestParameters', 'resources']]


display(df_cleaned)

score             eventTime  \
0  18.184576  2025-03-03T15:16:30Z   
1  17.897140  2025-03-03T15:16:30Z   
2  17.475600  2025-03-03T15:13:31Z   
3  16.519758  2025-03-03T13:27:42Z   
4  15.838386  2025-03-03T13:48:47Z   

                                        user_arn  \
0  arn:aws:iam::538539780105:user/lambda_s3_user   
1  arn:aws:iam::538539780105:user/lambda_s3_user   
2  arn:aws:iam::538539780105:user/lambda_s3_user   
3                 arn:aws:iam::538539780105:root   
4                 arn:aws:iam::538539780105:root   

                          eventName     awsRegion sourceIPAddress  \
0                         GetObject  eu-central-1    95.168.107.4   
1                        HeadObject  eu-central-1    95.168.107.4   
2                       ListObjects  eu-central-1    95.168.107.4   
3  GetBucketObjectLockConfiguration  eu-central-1    95.168.107.4   
4  GetBucketObjectLockConfiguration  eu-central-1    95.168.107.4   

                                                                                                                                                             userAgent  \
0  [aws-cli/1.36.17 md/Botocore#1.35.71 ua/2.0 os/linux#5.15.0-86-generic md/arch#x86_64 lang/python#3.10.12 md/pyimpl#CPython cfg/retry-mode#legacy botocore/1.35.71]   
1  [aws-cli/1.36.17 md/Botocore#1.35.71 ua/2.0 os/linux#5.15.0-86-generic md/arch#x86_64 lang/python#3.10.12 md/pyimpl#CPython cfg/retry-mode#legacy botocore/1.35.71]   
2  [aws-cli/1.36.17 md/Botocore#1.35.71 ua/2.0 os/linux#5.15.0-86-generic md/arch#x86_64 lang/python#3.10.12 md/pyimpl#CPython cfg/retry-mode#legacy botocore/1.35.71]   
3                                                              [Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36]   
4                                                              [Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36]   

                                                                                                                                                  requestParameters  \
0                                       {'bucketName': 'bsides-zagreb', 'Host': 'bsides-zagreb.s3.eu-central-1.amazonaws.com', 'key': 'super_secret_text_file.txt'}   
1                                       {'bucketName': 'bsides-zagreb', 'Host': 'bsides-zagreb.s3.eu-central-1.amazonaws.com', 'key': 'super_secret_text_file.txt'}   
2  {'list-type': '2', 'bucketName': 'bsides-zagreb', 'encoding-type': 'url', 'prefix': '', 'delimiter': '/', 'Host': 'bsides-zagreb.s3.eu-central-1.amazonaws.com'}   
3                                                                       {'bucketName': 'bsides-zagreb', 'object-lock': '', 'Host': 's3.eu-central-1.amazonaws.com'}   
4                                                                       {'bucketName': 'bsides-zagreb', 'object-lock': '', 'Host': 's3.eu-central-1.amazonaws.com'}   

                                                                                                                                                                                      resources  
0  [{'type': 'AWS::S3::Object', 'ARN': 'arn:aws:s3:::bsides-zagreb/super_secret_text_file.txt'}, {'accountId': '538539780105', 'type': 'AWS::S3::Bucket', 'ARN': 'arn:aws:s3:::bsides-zagreb'}]  
1  [{'type': 'AWS::S3::Object', 'ARN': 'arn:aws:s3:::bsides-zagreb/super_secret_text_file.txt'}, {'accountId': '538539780105', 'type': 'AWS::S3::Bucket', 'ARN': 'arn:aws:s3:::bsides-zagreb'}]  
2                      [{'type': 'AWS::S3::Object', 'ARNPrefix': 'arn:aws:s3:::bsides-zagreb/'}, {'accountId': '538539780105', 'type': 'AWS::S3::Bucket', 'ARN': 'arn:aws:s3:::bsides-zagreb'}]  
3                                                                                               [{'accountId': '538539780105', 'type': 'AWS::S3::Bucket', 'ARN': 'arn:aws:s3:::bsides-zagreb'}]  
4                                                       

---
### 2. See what else we can find related to before mentioned user **lambda_s3_user**
- By user name we can't really find connection to previous actions, this indicates that attacker escalated to this user.
- We still have IP address that commands came from. From username we can assume that is is somehow related to lambda functions so we can check the actions performed on any lambda function.
- Lambda functions can often leak hardcoded credentials in code or env variables.
- Here we can see some problems with getting high score results for get function query since the event name for it is actually GetFunction20150331v2.
- Listfunction api calls are executed via CLI using assumed role **S3_Lambda**

In [69]:


query = "get lambda function user agent is botocore"

#query = "actions from aws cli and source IP is 95.168.107.4" Bad results, doesn't really filter out stuff from this IP




response = client.search(
    index=".ds-logs-aws.cloudtrail-default-2025.03.04-000016",
    size=25,
    query={
      "sparse_vector": {
        "field": "message_elser_v2",
         "inference_id": ".elser_model_2_linux-x86_64",
          "query": query

  }
},
)

results = []

for hit in response["hits"]["hits"]:
    
    event_original = json.loads(hit["_source"]["event"]["original"])
    event_original['score'] = hit["_score"]
    results.append(event_original)


df_all = pd.DataFrame(results)
df_all['user_arn'] = df_all['userIdentity'].apply(lambda x: x['arn'] if 'arn' in x.keys() else x)
df_cleaned = df_all[['score','eventTime','user_arn', 'eventName', 'awsRegion', "sourceIPAddress", 'userAgent', 'requestParameters', "responseElements"]]

pd.set_option('display.max_colwidth', None)

display(df_cleaned)

score             eventTime  \
0   12.608977  2025-03-03T15:13:09Z   
1   11.921554  2025-03-03T15:16:30Z   
2   11.826011  2025-03-03T15:16:30Z   
3   11.798935  2025-03-03T15:09:53Z   
4   11.239194  2025-03-03T15:13:31Z   
5   11.099090  2025-03-03T15:07:03Z   
6   10.832518  2025-03-03T15:07:29Z   
7   10.827883  2025-03-03T15:11:31Z   
8   10.798711  2025-03-03T15:07:45Z   
9   10.351681  2025-03-03T15:11:40Z   
10   9.355085  2025-03-03T13:54:25Z   
11   9.149727  2025-03-03T13:54:23Z   
12   8.879747  2025-03-03T13:52:01Z   
13   8.853580  2025-03-03T13:53:41Z   
14   8.731368  2025-03-03T13:52:03Z   
15   8.681765  2025-03-03T14:31:13Z   
16   8.681595  2025-03-03T10:30:25Z   
17   8.632763  2025-03-03T13:52:05Z   
18   8.530870  2025-03-03T13:54:22Z   
19   8.504450  2025-03-03T15:08:55Z   
20   8.381065  2025-03-03T13:54:31Z   
21   8.314053  2025-03-03T15:08:17Z   
22   8.312501  2025-03-03T13:55:25Z   
23   8.262690  2025-03-03T14:31:13Z   
24   8.261606  2025-03-03T13:56:32Z   

                                                                user_arn  \
0                          arn:aws:iam::538539780105:user/lambda_s3_user   
1                          arn:aws:iam::538539780105:user/lambda_s3_user   
2                          arn:aws:iam::538539780105:user/lambda_s3_user   
3   arn:aws:sts::538539780105:assumed-role/S3_Lambda/i-0905d72baf3abed14   
4                          arn:aws:iam::538539780105:user/lambda_s3_user   
5   arn:aws:sts::538539780105:assumed-role/S3_Lambda/i-0905d72baf3abed14   
6   arn:aws:sts::538539780105:assumed-role/S3_Lambda/i-0905d72baf3abed14   
7   arn:aws:sts::538539780105:assumed-role/S3_Lambda/i-0905d72baf3abed14   
8   arn:aws:sts::538539780105:assumed-role/S3_Lambda/i-0905d72baf3abed14   
9   arn:aws:sts::538539780105:assumed-role/S3_Lambda/i-0905d72baf3abed14   
10                                        arn:aws:iam::538539780105:root   
11                                        arn:aws:iam::538539780105:root   
12                                        arn:aws:iam::538539780105:root   
13                   {'type': 'AWSService', 'invokedBy': 'AWS Internal'}   
14                                        arn:aws:iam::538539780105:root   
15  arn:aws:sts::538539780105:assumed-role/S3_Lambda/i-0905d72baf3abed14   
16                                 arn:aws:iam::538539780105:user/fzagar   
17                                        arn:aws:iam::538539780105:root   
18                                        arn:aws:iam::538539780105:root   
19                   {'type': 'AWSService', 'invokedBy': 'AWS Internal'}   
20                   {'type': 'AWSService', 'invokedBy': 'AWS Internal'}   
21                                        arn:aws:iam::538539780105:root   
22                   {'type': 'AWSService', 'invokedBy': 'AWS Internal'}   
23  arn:aws:sts::538539780105:assumed-role/S3_Lambda/i-0905d72baf3abed14   
24                   {'type': 'AWSService', 'invokedBy': 'AWS Internal'}   

                             eventName     awsRegion sourceIPAddress  \
0                    GetCallerIdentity     us-east-1    95.168.107.4   
1                            GetObject  eu-central-1    95.168.107.4   
2                           HeadObject  eu-central-1    95.168.107.4   
3                GetFunction20150331v2  eu-central-1    95.168.107.4   
4                          ListObjects  eu-central-1    95.168.107.4   
5                    GetCallerIdentity     us-east-1    95.168.107.4   
6                ListFunctions20150331  eu-central-1    95.168.107.4   
7                          ListBuckets  eu-central-1    95.168.107.4   
8                          ListBuckets  eu-central-1    95.168.107.4   
9                          ListObjects  eu-central-1    95.168.107.4   
10              CreateFunction20150331  eu-central-1    95.168.107.4   
11              CreateFunction20150331  eu-central-1    95.168.107.4   
12              CreateFunction20150331  eu-central-1    95.168.107.4   
13  GetF

---
### 4. Investigate events related to instance i-0905d72baf3abed14
    From role ARN arn:aws:sts::538539780105:assumed-role/S3_Lambda/*i-0905d72baf3abed14* we can conclude that this was role attached to an EC2 instance with ID i-0905d72baf3abed14. Requests are coming from public IP and not directly from instance which leads us to assuming temporary credentials related to this EC2 instance have been stolen.

In [6]:
query = "meta data iam S3_lambda"


response = client.search(
    index="logs-aws.cloudtrail*",
    size=15,
    query={
      "sparse_vector": {
        "field": "message_elser_v2",
         "inference_id": ".elser_model_2_linux-x86_64",
          "query": query

  }
},
)

results = []

for hit in response["hits"]["hits"]:

    event = json.loads(hit["_source"]["event"]["original"])

        
    event["score"] = float(hit["_score"])
    results.append(event)

df_all = pd.DataFrame(results)
df_cleaned= df_all[['score','eventTime','userIdentity', 'eventName', 'awsRegion', "sourceIPAddress", 'userAgent', 'requestParameters']]

pd.set_option('display.max_colwidth', None)

display(df_cleaned)

score             eventTime  \
0   15.591262  2025-03-03T15:16:30Z   
1   15.126850  2025-03-03T15:13:09Z   
2   15.039971  2025-03-03T15:16:30Z   
3   15.037180  2025-03-03T13:52:03Z   
4   14.618403  2025-03-03T13:52:01Z   
5   14.592430  2025-03-03T13:52:09Z   
6   14.501382  2025-03-03T15:13:31Z   
7   14.172089  2025-03-03T13:56:31Z   
8   14.035677  2025-03-03T13:52:05Z   
9   13.835537  2025-03-03T13:53:40Z   
10  13.402768  2025-03-03T15:08:54Z   
11  13.001299  2025-03-03T13:54:30Z   
12  12.648346  2025-03-03T13:55:24Z   
13  12.477847  2025-03-03T14:31:13Z   
14  12.357701  2025-03-03T13:54:25Z   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     userIdentity  \
0                                                                                                                                                                                                                                                                                                                                           {'type': 'IAMUser', 'principalId': 'AIDAX2Y3Y5AE7KVE2CM3O', 'arn': 'arn:aws:iam::538539780105:user/lambda_s3_user', 'accountId': '538539780105', 'accessKeyId': 'AKIAX2Y3Y5AESBKH7OGC', 'userName': 'lambda_s3_user'}   
1                                                                                                                                                                                                                                                                                                                                           {'type': 'IAMUser', 'principalId': 'AIDAX2Y3Y5AE7KVE2CM3O', 'arn': 'arn:aws:iam::538539780105:user/lambda_s3_user', 'accountId': '538539780105', 'accessKeyId': 'AKIAX2Y3Y5AESBKH7OGC', 'userName': 'lambda_s3_user'}   
2                                                                                                                                                                                                                                                                                                                                           {'type': 'IAMUser', 'principalId': 'AIDAX2Y3Y5AE7KVE2CM3O', 'arn': 'arn:aws:iam::538539780105:user/lambda_s3_user', 'accountId': '538539780105', 'accessKeyId': 'AKIAX2Y3Y5AESBKH7OGC', 'userName': 'lambda_s3_user'}   
3                                                                                                                                                                                                                                                                                            {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AE3LOFEY7L', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
4                                                                                                                                                                                                                                                                                            {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AE3LOFEY7L', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
5                                                                                        

---
### 5. Pivot to Application load balancer logs for more info

- Cloudtrail by itself doesn't log requests to instance metadata service
- We can ry to connect API explitation attempts to the compromised instance
- From ALB logs we can confirm that attacker abused instance metadata service to obtain credentials to impersonate role S3_lambda.

In [12]:
query = "meta data iam S3_lambda"


response = client.search(
    index="logs-aws.elb_logs-default",
    size=15,
    query={
      "sparse_vector": {
        "field": "message_elser_v2",
         "inference_id": ".elser_model_2_linux-x86_64_ingest",
          "query": query
  }
},
)

for hit in response["hits"]["hits"]:
        
    print(f"Score:{hit['_score']}")
    print(f"Message: {hit['_source']['event']['original']}",end=f"\n\n{120*'-'}\n\n")
    

Score:9.403352
Message: http 2025-03-03T14:52:17.471449Z app/vuln-app-alb/72fb0c71304e1795 95.168.107.4:29916 10.0.0.78:9000 0.001 0.006 0.000 200 200 361 1817 "GET http://vuln-app-alb-404212911.eu-central-1.elb.amazonaws.com:8234/get?url=http%3A%2F%2F169.254.169.254%2Flatest%2Fmeta-data%2Fiam%2Fsecurity-credentials%2FS3_Lambda HTTP/1.1" "HTTPie/3.2.4" - - arn:aws:elasticloadbalancing:eu-central-1:538539780105:targetgroup/vuln-app-targets/28bc56e20e82ee3f "Root=1-67c5c221-5e9ceb4670872dca2cfc87fa" "-" "-" 0 2025-03-03T14:52:17.464000Z "forward" "-" "-" "10.0.0.78:9000" "200" "-" "-" TID_d29df88a9bdf6747aa8fdc8672cd8e73

------------------------------------------------------------------------------------------------------------------------

Score:4.930862
Message: http 2025-03-03T14:27:42.122817Z app/vuln-app-alb/72fb0c71304e1795 95.168.107.4:33746 10.0.0.78:9000 0.001 0.002 0.000 405 405 320 208 "GET http://vuln-app-alb-404212911.eu-central-1.elb.amazonaws.com:8234/put?url=http%3A%2F%2

---
### 6. Find more about the connection with specific instance by querying for target group
    If we can find event of target group creation and registration we can confirm that instance belongs to vulnerable target group and was the entry point for the attacker.

In [20]:
query = "target group vuln-app-targets i-0905d72baf3abed14"


response = client.search(
    index="logs-aws.cloudtrail*",
    size=15,
    query={
      "sparse_vector": {
        "field": "message_elser_v2",
         "inference_id": ".elser_model_2_linux-x86_64",
          "query": query

  }
},
)

results = []

for hit in response["hits"]["hits"]:

    event = json.loads(hit["_source"]["event"]["original"])

        
    event["score"] = float(hit["_score"])
    results.append(event)  

df_all = pd.DataFrame(results)
df_cleaned= df_all[['score','eventTime','userIdentity', 'eventName', 'awsRegion', "sourceIPAddress", 'userAgent', 'requestParameters']]


pd.set_option('display.max_colwidth', None)

display(df_cleaned)

score             eventTime  \
0   18.524036  2025-03-03T07:56:13Z   
1   18.341840  2025-03-03T12:52:55Z   
2   18.207516  2025-03-03T11:33:36Z   
3   18.010761  2025-03-03T12:56:24Z   
4   17.958190  2025-03-03T12:59:21Z   
5   17.755571  2025-03-03T07:53:14Z   
6   17.286861  2025-03-03T12:57:36Z   
7   17.251030  2025-03-03T12:59:42Z   
8   17.240840  2025-03-03T12:49:36Z   
9   17.203327  2025-03-03T11:39:26Z   
10  17.141563  2025-03-03T12:47:32Z   
11  17.098251  2025-03-03T11:37:59Z   
12  17.096430  2025-03-03T08:17:50Z   
13  17.079070  2025-03-03T12:47:28Z   
14  16.969324  2025-03-03T12:51:00Z   

                                                                                                                                                                                                                                                            userIdentity  \
0   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AEWYXOCTBN', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
1   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AEZI24OWWS', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
2   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AE3RJTLRG4', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
3   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AEVIVCRLH7', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
4   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AEVIVCRLH7', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
5   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AEWYXOCTBN', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
6   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AEZI24OWWS', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
7   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AEVIVCRLH7', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
8   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AERUJIFQTM', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
9   {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AE3RJTLRG4', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
10  {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AERUJIFQTM', 'sessionContext': {'attributes': {'creationDate': '2025-03-03T07:49:19Z', 'mfaAuthenticated': 'true'}}}   
11  {'type': 'Root', 'principalId': '538539780105', 'arn': 'arn:aws:iam::538539780105:root', 'accountId': '538539780105', 'accessKeyId': 'ASIAX2Y3Y5AE3RJTLRG4', 'sessi